test/val/val_id_test.cpp

```diff
@@ -527,12 +555,13 @@ class OpTypeArrayLengthTest
   ~OpTypeArrayLengthTest() { spvDiagnosticDestroy(diagnostic_); }
 
   // Runs spvValidate() on v, printing any errors via spvDiagnosticPrint().
-  spv_result_t Val(const SpirvVector& v) {
+  spv_result_t Val(const SpirvVector& v, const char* expected_err = "") {
```

Comment:
In this case it's better to make this const std::string&  parameter.
You never allow it to be nullptr.  And you construct the std::string later anyway in the error case.  Hm. Are you trying to save the construction time?  I'm not worried about that kind of optimization in test code. In test code the balance shifts toward clarity over a small optimization like this.